l1-contracts/contracts/bridge/L1NativeTokenVault.sol

```diff
@@ -0,0 +1,223 @@
+            return abi.encode(name, symbol, decimals); // when depositing eth to a non-eth based chain it is an ERC20
+        }
+
+        (, bytes memory data1) = _token.staticcall(abi.encodeCall(IERC20Metadata.name, ()));
+        (, bytes memory data2) = _token.staticcall(abi.encodeCall(IERC20Metadata.symbol, ()));
+        (, bytes memory data3) = _token.staticcall(abi.encodeCall(IERC20Metadata.decimals, ()));
+        return abi.encode(data1, data2, data3);
+    }
+
+    /* solhint-disable no-unused-vars */
+    function bridgeMint(uint256 _chainId, bytes32 _assetInfo, bytes calldata _data) external payable override {
+        // if (!hyperbridgingEnabled[_chainId]) { // ToDo: add logic & uncomment
+        //     // Add back
+        //     // Check that the chain has sufficient balance
+        //     require(chainBalance[_chainId][l1Token] >= _amount, "NTV not enough funds 2"); // not enough funds
+        //     chainBalance[_chainId][l1Token] -= _amount;
+        // }
+        address l1Token = tokenAddress[_assetInfo];
+        (uint256 _amount, address l1Receiver) = abi.decode(_data, (uint256, address));
+        if (l1Token == ETH_TOKEN_ADDRESS) {
+            bool callSuccess;
+            // Low-level assembly call, to avoid any memory copying (save gas)
+            assembly {
+                callSuccess := call(gas(), l1Receiver, _amount, 0, 0, 0, 0)
+            }
+            require(callSuccess, "NTV: withdraw failed");
+        } else {
+            // Withdraw funds
+            IERC20(l1Token).safeTransfer(l1Receiver, _amount);
+        }
+    }
+
+    function bridgeClaimFailedBurn(
+        uint256 _chainId,
+        bytes32 _assetInfo,
+        address,
+        bytes calldata _data
+    ) external payable override {
+        (uint256 _amount, address _depositSender) = abi.decode(_data, (uint256, address));
+        address l1Token = tokenAddress[_assetInfo];
+        require(_amount > 0, "y1");
+
+        if (l1Token == ETH_TOKEN_ADDRESS) {
+            bool callSuccess;
+            // Low-level assembly call, to avoid any memory copying (save gas)
+            assembly {
+                callSuccess := call(gas(), _depositSender, _amount, 0, 0, 0, 0)
+            }
+            require(callSuccess, "NTV: claimFailedDeposit failed");
+        } else {
+            IERC20(l1Token).safeTransfer(_depositSender, _amount);
+            // Note we don't allow weth deposits anymore, but there might be legacy weth deposits.
+            // until we add Weth bridging capabilities, we don't wrap/unwrap weth to ether.
+        }
+
+        if (L1_SHARED_BRIDGE.hyperbridgingEnabled(_chainId)) {
+            // check that the chain has sufficient balance
+            require(chainBalance[_chainId][l1Token] >= _amount, "NTV n funds");
+            chainBalance[_chainId][l1Token] -= _amount;
+        }
+    }
+
+    function getAssetInfoFromLegacy(address _l1TokenAddress) public view override returns (bytes32) {
```

Comment:
Do you view conducting a bigger migration (to assign each old asset an asset info), but removing this legacy method as feasible?

Reply:
I think it is a good idea, we could perform this from backend, but it would result in cleaner logic with less conditional statements in sharedBridge

Reply:
If we want to be compatible with the old SDK then we will still need to have some functionality though for new tokens.